src/App.js

```diff
@@ -32,14 +27,21 @@ function App() {
         <Switch>
           <Route path="/signUp" exact component={SignUp} />
           <Route path="/forgotPassword" exact component={ForgotPassword} />
-          <Route path="/login" exact component={Login} />
+          <Route path="/login" exact render={() =>
+            <Login
+              setSignedIn={setSignedIn}
+              setUser={setUser}
+            />
+          } />
```

Comment:
The two set__ functions probably don't need to get passed. The ``onAuthStateChanged`` listener will be triggered whenever you log in. This also allows for automatic login, which would be a killer feature. Fire off the set events when ``user`` is passed in the Firebase callback.

Reply:
Automatic login would be great! Thanks for pointing that out, I didn't realize that `onAuthStateChanged` listener would be called for every login and logout. I changed it so that the set_ functions aren't passed to the Login or Validator components, and they are instead updated in the listener's callback in App.js itself. The app's behavior still seems the same.